pkg/storage/filesystem/btrfs.go

```diff
@@ -314,7 +325,19 @@ func (p *btrfsPool) AddVolume(name string) (Volume, error) {
 }
 
 func (p *btrfsPool) removeVolume(root string) error {
-	return p.utils.SubvolumeRemove(context.Background(), root)
+	ctx := context.Background()
```

Comment:
I would actually remove the subvol first before the qgroup. Because if the subvolume deletion failed for any reason, we don't leave a subvolume with no size limit.